tests/catch/unit/module/hip_module_launch_kernel_common.hh

```diff
@@ -0,0 +1,262 @@
+                                   unsigned int gridDimZ, unsigned int blockDimX,
+                                   unsigned int blockDimY, unsigned int blockDimZ) {
+    hipFunction_t f = GetKernel(mg.module(), "NOPKernel");
+    HIP_CHECK(func(f, gridDimX, gridDimY, gridDimZ, blockDimX, blockDimY, blockDimZ, 0, nullptr,
+                   nullptr, nullptr, nullptr, nullptr, 0u));
+    HIP_CHECK(hipDeviceSynchronize());
+  };
+
+  SECTION("gridDimX == maxGridDimX") {
+    const unsigned int x = GetDeviceAttribute(0, hipDeviceAttributeMaxGridDimX);
+    LaunchNOPKernel(x, 1, 1, 1, 1, 1);
+  }
+
+  SECTION("gridDimY == maxGridDimY") {
+    const unsigned int y = GetDeviceAttribute(0, hipDeviceAttributeMaxGridDimY);
+    LaunchNOPKernel(1, y, 1, 1, 1, 1);
+  }
+
+  SECTION("gridDimZ == maxGridDimZ") {
+    const unsigned int z = GetDeviceAttribute(0, hipDeviceAttributeMaxGridDimZ);
+    LaunchNOPKernel(1, 1, z, 1, 1, 1);
+  }
+
+  SECTION("blockDimX == maxBlockDimX") {
+    const unsigned int x = GetDeviceAttribute(0, hipDeviceAttributeMaxBlockDimX);
+    LaunchNOPKernel(1, 1, 1, x, 1, 1);
+  }
+
+  SECTION("blockDimY == maxBlockDimY") {
+    const unsigned int y = GetDeviceAttribute(0, hipDeviceAttributeMaxBlockDimY);
+    LaunchNOPKernel(1, 1, 1, 1, y, 1);
+  }
+
+  SECTION("blockDimZ == maxBlockDimZ") {
+    const unsigned int z = GetDeviceAttribute(0, hipDeviceAttributeMaxBlockDimZ);
+    LaunchNOPKernel(1, 1, 1, 1, 1, z);
+  }
+}
+
+template <ExtModuleLaunchKernelSig* func> void ModuleLaunchKernelNegativeParameters() {
+  hipFunction_t f = GetKernel(mg.module(), "NOPKernel");
+
+// Disabled on AMD due to defect
+#if HT_NVIDIA
+  SECTION("f == nullptr") {
+    HIP_CHECK_ERROR(
+        func(nullptr, 1, 1, 1, 1, 1, 1, 0, nullptr, nullptr, nullptr, nullptr, nullptr, 0u),
+        hipErrorInvalidResourceHandle);
+  }
+#endif
+
+  SECTION("gridDimX == 0") {
+    HIP_CHECK_ERROR(func(f, 0, 1, 1, 1, 1, 1, 0, nullptr, nullptr, nullptr, nullptr, nullptr, 0u),
+                    hipErrorInvalidValue);
+  }
+
+  SECTION("gridDimY == 0") {
+    HIP_CHECK_ERROR(func(f, 1, 0, 1, 1, 1, 1, 0, nullptr, nullptr, nullptr, nullptr, nullptr, 0u),
+                    hipErrorInvalidValue);
+  }
+
+  SECTION("gridDimZ == 0") {
+    HIP_CHECK_ERROR(func(f, 1, 1, 0, 1, 1, 1, 0, nullptr, nullptr, nullptr, nullptr, nullptr, 0u),
+                    hipErrorInvalidValue);
+  }
+
+  SECTION("blockDimX == 0") {
+    HIP_CHECK_ERROR(func(f, 1, 1, 1, 0, 1, 1, 0, nullptr, nullptr, nullptr, nullptr, nullptr, 0u),
+                    hipErrorInvalidValue);
+  }
+
+  SECTION("blockDimY == 0") {
+    HIP_CHECK_ERROR(func(f, 1, 1, 1, 1, 0, 1, 0, nullptr, nullptr, nullptr, nullptr, nullptr, 0u),
+                    hipErrorInvalidValue);
+  }
+
+  SECTION("blockDimZ == 0") {
+    HIP_CHECK_ERROR(func(f, 1, 1, 1, 1, 1, 0, 0, nullptr, nullptr, nullptr, nullptr, nullptr, 0u),
+                    hipErrorInvalidValue);
+  }
+
+// Disabled on AMD due to defect
+#if HT_NVIDIA
```

Comment:
is there a ticket?

Reply:
There is. I've updated the PR with the ticket IDs added in the comments.